feconf.py

```diff
@@ -30,8 +31,9 @@
 # Whether we should serve the development or production experience.
 if PLATFORM == 'gae':
     DEV_MODE = (
-        not os.environ.get('SERVER_SOFTWARE')
-        or os.environ['SERVER_SOFTWARE'].startswith('Development'))
+        (not os.environ.get('SERVER_SOFTWARE')
+         or os.environ['SERVER_SOFTWARE'].startswith('Development'))
+        and os.environ.get('MINIFICATION') is None)
```

Comment:
Logging the env variable gives a 'None' value. GAE resets all env variables and sets the ones it needs.